src/mscorlib/src/System/Runtime/Intrinsics/X86/Sse2.cs

```diff
@@ -971,47 +972,191 @@ public static class Sse2
+        public static Vector128<uint> SetVector128(uint e3, uint e2, uint e1, uint e0)
+        {
+            // TODO-CQ Optimize algorithm choice based on benchmarks
+
+            // Zero vector and load e2 et index 0
+            Vector128<uint> e2Vector = ConvertScalarToVector128UInt32(e2);
+            Vector128<uint> e1Vector = ConvertScalarToVector128UInt32(e1);
+            Vector128<uint> e0Vector = ConvertScalarToVector128UInt32(e0);
+            // Create { -- -- e2 e0 }
+            e0Vector = UnpackLow(e0Vector, e2Vector);
+            e2Vector = ConvertScalarToVector128UInt32(e3);
+            // Create { -- -- e3 e1 }
+            e1Vector = UnpackLow(e1Vector, e2Vector);
+            // Create { e3 e2 e1 e0 } and return result
+            return UnpackLow(e0Vector, e1Vector);
+        }
         /// <summary>
         /// __m128i _mm_set_epi64x (__int64 e1, __int64 e0)
         ///   HELPER
         /// </summary>
-        public static Vector128<long> SetVector128(long e1, long e0) => SetVector128(e1, e0);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<long> SetVector128(long e1, long e0)
+        {
+            // TODO-CQ Optimize algorithm choice based on benchmarks
+
+            // Zero vector and load e1 et index 0
+            Vector128<long> e1Vector = ConvertScalarToVector128Int64(e1);
+            Vector128<long> e0vector = ConvertScalarToVector128Int64(e0);
+            // Create { e1 e0 } and return result
+            return UnpackLow(e0vector, e1Vector);
+        }
         /// <summary>
         /// __m128i _mm_set_epi64x (__int64 e1, __int64 e0)
         ///   HELPER
         /// </summary>
-        public static Vector128<ulong> SetVector128(ulong e1, ulong e0) => SetVector128(e1, e0);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<ulong> SetVector128(ulong e1, ulong e0)
+        {
+            // TODO-CQ Optimize algorithm choice based on benchmarks
+
+            // Zero vector and load e1 et index 0
+            Vector128<ulong> e1vector = ConvertScalarToVector128UInt64(e1);
+            Vector128<ulong> e0Vector = ConvertScalarToVector128UInt64(e0);
+            // Create { e1 e0 } and return result
+            return UnpackLow(e0Vector, e1vector);
+        }
         /// <summary>
         /// __m128d _mm_set_pd (double e1, double e0)
         ///   HELPER
         /// </summary>
-        public static Vector128<double> SetVector128(double e1, double e0) => SetVector128(e1, e0);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static unsafe Vector128<double> SetVector128(double e1, double e0)
+        {
+            // TODO-CQ Optimize algorithm choice based on benchmarks
+
+            // Zero vector and load e1 et index 0
+            Vector128<double> e1Vector = LoadScalarVector128(&e1);
+            Vector128<double> e0Vector = LoadScalarVector128(&e0);
```

Comment:
I do not like these memory round-trips (arg-reg -> memory-> vector reg), `SetScalarVector128<T>` should be better.

Reply:
Yes, you are right.